### PR TITLE
MM-41988 Updated links to legacy domain about.mm.com

### PR DESCRIPTION
**Pull request by cwarnermm**
#### Summary
Updated in-product links to legacy domain about.mattermost.com.

#### Tickets
https://mattermost.atlassian.net/browse/MM-41988

#### Release Note
```release-note
Updated in-product links to legacy domain about.mattermost.com
```

### CODE REVIEW

File: detox/e2e/support/server_api/default_config.json [hunk 1]
```diff
@@ -233,8 +233,8 @@
     },
     "NativeAppSettings": {
         "AppDownloadLink": "https://mattermost.com/download/#mattermostApps",
-        "AndroidAppDownloadLink": "https://about.mattermost.com/mattermost-android-app/",
-        "IosAppDownloadLink": "https://about.mattermost.com/mattermost-ios-app/"
+        "AndroidAppDownloadLink": "https://play.google.com/store/apps/details?id=com.mattermost.rn",
+        "IosAppDownloadLink": "https://apps.apple.com/us/app/mattermost/id1257222717"
```
Comment (enahum) [inline detox/e2e/support/server_api/default_config.json:237]: Why this change? I believe this is handled by a redirect?

**Review by enahum — CHANGES_REQUESTED** [review]

File: detox/e2e/support/server_api/default_config.json (hunk 1 above)
Comment (cwarnermm) [reply detox/e2e/support/server_api/default_config.json:237]: @enahum - Marketing is sunsetting a number of existing domains, and I'm working with Marketing to ensure that all links to the sunsetted domains are updated, including docs & product.

Yes, there are going to be redirects in place; however, we're formally updating all of the links so that we're no longer relying on those redirects being in place. There are a LOT of old links, like these two, across the product code.

Do you have a concern about updating these links from a development perspective? Or would changing these links cause downstream impact I haven't accounted for?

Comment (enahum) [reply detox/e2e/support/server_api/default_config.json:237]: no worries at all,.. just wanted to be sure

**Review by enahum — APPROVED** [review]

File: assets/base/config.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
     "DefaultServerUrl": "",
     "TestServerUrl": "http://localhost:8065",
     "ShowErrorsList": false,
-    "EELearnURL": "about.mattermost.com",
+    "EELearnURL": "mattermost.com",
     "TeamEditionLearnURL": "mattermost.org",
-    "AboutTeamURL": "http://www.mattermost.org/",
+    "AboutTeamURL": "https://mattermost.com/",
     "AboutEnterpriseURL": "https://mattermost.com/",
     "PlatformNoticeURL": "https://mattermost.com/platform-notice-txt/",
```
Comment (josephbaylon) [inline assets/base/config.json:9]: Hi @cwarnermm Not sure if this is within the scope of your changes but the `PlatformNoticeURL` doesn't seem to have the proper url (https://mattermost.com/platform-notice-txt/ is a non-existent page)

**Review by josephbaylon — CHANGES_REQUESTED** [review]

File: assets/base/config.json (hunk 2 above)
Comment (cwarnermm) [reply assets/base/config.json]: @josephbaylon - It is and I've made that update. Thank you for flagging it!

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @cwarnermm !!
